src/core/lib/surface/call.cc

```diff
@@ -1035,9 +1035,15 @@ static void recv_trailing_filter(void* args, grpc_metadata_batch* b,
         grpc_get_status_code_from_metadata(b->idx.named.grpc_status->md);
     grpc_error* error = GRPC_ERROR_NONE;
     if (status_code != GRPC_STATUS_OK) {
-      error = grpc_error_set_int(
-          GRPC_ERROR_CREATE_FROM_STATIC_STRING("Error received from peer"),
-          GRPC_ERROR_INT_GRPC_STATUS, static_cast<intptr_t>(status_code));
+      char* peer_msg = nullptr;
+      char* peer = grpc_call_get_peer(call);
+      gpr_asprintf(&peer_msg, "Error received from peer %s",
+                   grpc_call_get_peer(call));
```

Comment:
This needs to use `peer` instead of calling `grpc_call_get_peer()` again, or else it's a memory leak.

Reply:
I wasn't expecting reviews that soon. I was being lazy and actually wanted to use the github tests on github to verify that the peer is being printed properly :)

Reply:
and the peer string is printed properly :) Fixed it

Reply:
FWIW, when I create a PR that isn't ready for review, I typically prepend the subject with "WIP:" and add the work-in-progress label, just so it's clear.